src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -22,135 +104,791 @@ const ActorInit En_Pamera_InitVars = {
+    this->actor.world.pos.y += this->actor.gravity;
+}
+
+void EnPamera_LookDownWell(EnPamera* this) {
+    func_80BD93CC(this, 1, 1);
+    func_800BDC5C(&this->skelAnime, animations, 4);
+    this->actionFunc = func_80BD8F60;
+}
+
+void func_80BD8F60(EnPamera* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xA, 0x3000, 0x1000);
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        func_800BDC5C(&this->skelAnime, animations, 2);
+        this->actor.speedXZ = 3.0f;
+        func_80BD93CC(this, 0, 0);
+        func_80BD8D80(this);
+    }
+}
+
+void func_80BD8FF0(EnPamera* this) {
+    Vec3f pameraPos;
+    s16 pameraYaw;
+
+    this->pathIndex = this->pathPointsCount - 1;
+    Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->pathPoints[this->pathPointsCount - 1]);
+    Math_Vec3s_ToVec3f(&pameraPos, &this->pathPoints[this->pathPointsCount - 2]);
+    pameraYaw = Math_Vec3f_Yaw(&pameraPos, &this->actor.world.pos);
+    this->actor.shape.rot.y = pameraYaw;
+    this->actor.world.rot.y = pameraYaw;
+    func_800BDC5C(&this->skelAnime, animations, 3);
+    this->actionFunc = func_80BD909C;
+}
+
+void func_80BD909C(EnPamera* this, GlobalContext* globalCtx) {
+}
+
+void func_80BD90AC(EnPamera* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (Player_GetMask(globalCtx) != PLAYER_MASK_STONE_MASK && (this->actionFunc != func_80BD8758) &&
+        (this->actionFunc != func_80BD8964) && (this->actionFunc != func_80BD8A7C) &&
+        (this->actionFunc != func_80BD8F60) && ((this->actionFunc != func_80BD8B70) || (this->pathIndex != 0)) &&
+        ((this->actionFunc != func_80BD8DB0) || (this->actor.speedXZ != 3.0f)) &&
+        ((this->actor.xzDistToPlayer < 150.0f) ||
+         ((this->actionFunc == func_80BD909C) &&
+          (Math_Vec3f_DistXZ(&this->actor.home.pos, &player->actor.world.pos) < 200.0f)))) {
+        if ((ActorCutscene_GetCanPlayNext(this->cutscenes[1])) && ((this->cutscenes[1] != -1))) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutscenes[1], &this->actor);
+            func_800E02AC(Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(this->cutscenes[1])), this);
+            EnPamera_LookDownWell(this);
+        } else if (this->cutscenes[1] != -1) {
+            ActorCutscene_SetIntentToPlay(this->cutscenes[1]);
+        } else {
+            EnPamera_LookDownWell(this);
+        }
+    }
+}
+
+s32 func_80BD9234(EnPamera* this, GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_EXPLOSIVES].first;
+
+    while (actor != NULL) {
+        if ((actor->id == ACTOR_EN_BOM) && (Math_Vec3f_DistXZ(&this->actor.world.pos, &actor->world.pos) < 500.0f) &&
+            (((EnBom*)actor)->timer == 0)) {
+            return 1;
+        } else {
+            actor = actor->next;
+        }
+    }
+    return 0;
```

Comment:
```suggestion
    EnBom* bomb = (EnBom*)globalCtx->actorCtx.actorList[ACTORCAT_EXPLOSIVES].first;

    while (bomb != NULL) {
        if ((bomb->actor.id == ACTOR_EN_BOM) && (Math_Vec3f_DistXZ(&this->actor.world.pos, &bomb->actor.world.pos) < 500.0f) &&
            (bomb->timer == 0)) {
            return 1;
        } else {
            bomb->actor = bomb->actor->next;
        }
    }
    return 0;
```
What do you think about something like this?

Reply:
Also separate to the previous comment, this can return true/false

Reply:
Granted this is an argument of semantics, but I feel `actor` is more appropriate for the iteration, because you don't really have an `EnBom` pointer until its confirmed that it has the correct actor id. I don't have too strong an opinion about this though.

Reply:
I originally had something like this, but @hensldm's comment makes sense, I believe this is iterating through all the actors that are present in ikana valley (the doors, loading points, guays, and other objects). But it also does make sense that it is specifically a bomb actor since it specifically checks EnBom's timer and not anything in the Actor* struct.

Reply:
@hensldm argument is the exact reason actor should be preferred over en_bom. You don't have a pointer to en_bom until it's validated, and this function never validates it.